### ISSUE TITLE
ASAN:DEADLYSIGNAL

### ISSUE THREAD

**Issue by tianmai1** [michaelrsweet/codedoc#13]
./codedoc poc1
```
ASAN:DEADLYSIGNAL
=================================================================
==130545==ERROR: AddressSanitizer: SEGV on unknown address 0x000000000000 (pc 0x562719a5afc5 bp 0x7ffe435142a0 sp 0x7ffe435141b0 T0)
==130545==The signal is caused by a READ memory access.
==130545==Hint: address points to the zero page.
    #0 0x562719a5afc4 in write_html_body /home/tianmai/workspace/codedoc(复件)/codedoc.c:6042
    #1 0x562719a5a4b9 in write_html /home/tianmai/workspace/codedoc(复件)/codedoc.c:5867
    #2 0x562719a48194 in main /home/tianmai/workspace/codedoc(复件)/codedoc.c:625
    #3 0x7ff114394c86 in __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x21c86)
    #4 0x562719a465e9 in _start (/home/tianmai/workspace/codedoc(复件)/codedoc+0xe5e9)

AddressSanitizer can not provide additional info.
SUMMARY: AddressSanitizer: SEGV /home/tianmai/workspace/codedoc(复件)/codedoc.c:6042 in write_html_body
==130545==ABORTING
```
```
In file: /home/tianmai/workspace/codedoc(复件)/codedoc.c
   6037       else
   6038       {
   6039 	type   = mxmlFindElement(scut, scut, "type", NULL, NULL, MXML_DESCEND_FIRST);
   6040 	string = mxmlGetText(mxmlGetLastChild(type), NULL);
   6041 
 ► 6042         if (*string != '*')
   6043 	  putc(' ', out);
   6044 
   6045 	fprintf(out, "%s;\n", name);
   6046       }
   6047 
```
[poc1.zip](https://github.com/michaelrsweet/codedoc/files/12024320/poc1.zip)

**Comment by tianmai1** [comment]
version3.7

**Comment by michaelrsweet** [comment]
[master 686cd0c] Fix fuzzing bugs (Issue #13, Issue #14, Issue #15)
